hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1837,18 +1835,14 @@ def constrained_complex(draw):
             zi = draw(floats(-max_magnitude, max_magnitude, **allow_kw))
             rmax = cathetus(max_magnitude, zi)
         # Draw the real part from the allowed range given the imaginary part
-        if min_magnitude is None or math.fabs(zi) >= min_magnitude:
+        if min_magnitude == 0 or math.fabs(zi) >= min_magnitude:
             zr = draw(floats(None if rmax is None else -rmax, rmax, **allow_kw))
         else:
             zr = draw(floats(cathetus(min_magnitude, zi), rmax, **allow_kw))
         # Order of conditions carefully tuned so that for a given pair of
         # magnitude arguments, we always either draw or do not draw the bool
         # (crucial for good shrinking behaviour) but only invert when needed.
-        if (
-            min_magnitude is not None
-            and draw(booleans())
-            and math.fabs(zi) <= min_magnitude
-        ):
+        if min_magnitude != 0 and draw(booleans()) and math.fabs(zi) <= min_magnitude:
```

Comment:
```suggestion
        if min_magnitude > 0 and draw(booleans()) and math.fabs(zi) <= min_magnitude:
```

Equivalent, but doesn't require the reader to remember the lower bound.